docs: ordered list api bepaald acceptatiecriteria (#1284)
nl-design-system/candidate: packages/docs/ordered-list-docs/docs/acceptance-criteria.md

- De Ordered List komt standaard niet voor in de focusvolgorde van de pagina.

[Meer informatie over deze acceptatiecriteria lees je op de componentpagina op de website van NL Design System.](https://nldesignsystem.nl/ordered-list/)

## Acceptatiecriteria APIs van de component

### CSS API

- Block class: `nl-ordered-list`

- Element classes:
  - `nl-ordered-list__item`

>   - `nl-ordered-list__marker`

Reviewer: ```suggestion
  - `nl-ordered-list__item`
  - `nl-ordered-list__marker`
  - `nl-ordered-list__inline-marker`
```

Author: Verwerkt, en __marker los neergezet onder `Pseudo-element classes`

- Modifier classes:

>   - `nl-ordered-list--arabic`

Reviewer: Deze is denk ik niet nodig, we kunnen gewoon altijd `:lang(ar)` matchen

Author: Dit staat dan niet in de acceptatiecriteria, ik zat te denken om 'm wel hier ergens te vermelden. Wat dacht je van zoiets:

```

### CSS API

- Block class: `nl-ordered-list`

- Element classes:
  - `nl-ordered-list__item`
  - `nl-ordered-list__inline-marker`

- Pseudo-element classes:
  - `nl-ordered-list__marker` voor `::marker`

- Modifier classes:
  - geen

- HTML classes:
  - `nl-html--all`
  - `nl-html--ordered-list`

- Aanvullende selectors:
  - `:lang(ar)` voor Arabische nummering

```

- HTML classes:

>   - `nl-html`

Reviewer: ```suggestion
  - `nl-html--all`
```

Author: Verwerkt

  - `nl-html--ordered-list`

#### Geldige combinaties van CSS classes

| Configuratie                              | Geldig | Opmerking                                                      |
| ----------------------------------------- | :----: | -------------------------------------------------------------- |
| `nl-ordered-list`                         |   ✅   |                                                                |

> | `nl-ordered-list nl-ordered-list--arabic` |   ✅   | Forceert Arabische nummering ongeacht de browser instellingen. |

Reviewer: ```suggestion
```

Author: Verwerkt

| `nl-ordered-list__item`                   |   ✅   | Gebruik als child van `nl-ordered-list`.                       |

> | `nl-ordered-list__marker`                 |   ✅   | Gebruik een custom marker binnen een `nl-ordered-list__item`.  |

Reviewer: ```suggestion
| `nl-ordered-list__inline-marker`                 |   ✅   | Gebruik een custom marker binnen een `nl-ordered-list__item`.  |
```

Author: Verwerkt

Author: En overal 'custom marker' en 'marker' veranderd naar consistent 'inline marker'

| `restProps` | `OlHTMLAttributes<HTMLOListElement>` | -       | nee       | Overige standaard HTML-attributen en events worden doorgestuurd naar het `ol` HTML-element. |

### OrderedListItem

#### OrderedListItem extends

Extends `LiHTMLAttributes<HTMLLIElement>` voor standaard HTML-attributen en events van een HTML `li` element.

Alle standaard HTML-attributen en events worden doorgestuurd naar het onderliggende `li` HTML-element.

#### OrderedListItem properties

De volgende props worden expliciet ondersteund:

| Prop          | Type                              | Default | Verplicht | Omschrijving                                                                                |
| ------------- | --------------------------------- | ------- | --------- | ------------------------------------------------------------------------------------------- |
| `children`    | `ReactNode`                       | -       | nee       | De inhoud van de component.                                                                 |
| `marker`      | `ReactNode`                       | -       | nee       | De inhoud van de marker slot, weergegeven in een `span.nl-ordered-list__marker`.            |

> | `markerLabel` | `string`                          | -       | nee\*     | Toegankelijke naam voor de custom marker. Verplicht wanneer `marker` is opgegeven.          |

Reviewer: ```suggestion
| `inlineMarker`      | `ReactNode`                       | -       | nee       | De inhoud van de marker slot, weergegeven in een `span.nl-ordered-list__inline-marker`.            |
| `inlineMarkerLabel` | `string`                          | -       | nee\*     | Toegankelijke naam voor de custom marker. Verplicht wanneer `marker` is opgegeven.          |
```

Author: Verwerkt

- De Ordered List komt standaard niet voor in de focusvolgorde van de pagina.

[Meer informatie over deze acceptatiecriteria lees je op de componentpagina op de website van NL Design System.](https://nldesignsystem.nl/ordered-list/)

## Acceptatiecriteria APIs van de component

### CSS API

- Block class: `nl-ordered-list`

- Element classes:
  - `nl-ordered-list__item`
  - `nl-ordered-list__inline-marker`

- Pseudo-element classes:

>   - `nl-ordered-list__marker` voor `::marker`

Reviewer: Ik denk dat we hier geen class name voor nodig hebben, alleen de mixin

Author: Ik zal 'm renamen naar Pseudo-element mixins zoals we dinsdag hebben besproken

Author: Done!

- Modifier classes:

>   - geen

Reviewer: @eslook Ik heb voor modifiers nog even een issue aangemaakt

https://github.com/nl-design-system/candidate/issues/1306

Author: Top! Ik heb 'm op het bord gezet voor de aankomende prioritering en de juiste parent gegeven

- HTML classes:
  - `nl-html--all`
  - `nl-html--ordered-list`

- Aanvullende selectors:

>   - `:lang(ar)` voor Arabische nummering

Reviewer: ```suggestion
  - `:lang(ar)` op de Ordered List voor Arabische nummering
```

### OrderedListItem

#### OrderedListItem extends

Extends `LiHTMLAttributes<HTMLLIElement>` voor standaard HTML-attributen en events van een HTML `li` element.

Alle standaard HTML-attributen en events worden doorgestuurd naar het onderliggende `li` HTML-element.

#### OrderedListItem properties

De volgende props worden expliciet ondersteund:

| Prop          | Type                              | Default | Verplicht | Omschrijving                                                                                   |
| ------------- | --------------------------------- | ------- | --------- | ---------------------------------------------------------------------------------------------- |
| `children`    | `ReactNode`                       | -       | nee       | De inhoud van de component.                                                                    |
| `inlineMarker`      | `ReactNode`                       | -       | nee       | De inhoud van de inline marker slot, weergegeven in een `span.nl-ordered-list__inline-marker`. |

> | `inlineMarkerLabel` | `string`                          | -       | nee\*     | Toegankelijke naam voor de inline marker. Verplicht wanneer `inlineMarker` is opgegeven.             |

Reviewer: ```suggestion
| `inlineMarkerLabel` | `ReactNode`                          | -       | nee\*     | Toegankelijke naam voor de inline marker. Verplicht wanneer `inlineMarker` is opgegeven.             |
```

Ik ben trouwens bang dat als we een label instellen dat we ook een property moeten meegeven om een `id` attribuut te genereren als we geen hardcoded `useId()` willen hebben die problematisch is voor `"use client";`

Author: Wil je vanwege de id de inlineMarkerLabel een ReactNode maken, of is dat onafhankelijk omdat je wilt dat mensen markup mee mogen geven aan de inlineMarkerLabel? Mijn idee namelijk voor de id is dan dit:

```
<UnorderedList>
  <UnorderedListItem
    inlineMarker={<Icon />}
    inlineMarkerLabel="Een informatieve label"
    inlineMarkerLabelId="label-id"
  >
      Foo Bar
  </UnorderedListItem
</UnorderedList>
```

Waar dan uitkomt:
```
<ul>
  <li>
    <span aria-labelledby="label-id"><Icon/></span>
    <span id="label-id">Informatieve label</span>
    Foo Bar
  </li>
</ul>
```

En dan dit voor decoratieve label:

```
<UnorderedList>
  <UnorderedListItem
    inlineMarker={<Icon />}
  >
      Foo Bar
  </UnorderedListItem
</UnorderedList>
```

Waar dan uitkomt:
```
<ul>
  <li>
    <span aria-hidden="true"><Icon/></span>
    Foo Bar
  </li>
</ul>
```

Reviewer: Volgens mij gaat het hier om de `<ol>` niet de `<ul>`, en voor de Ordered List is er niet echt sprake van een decoratieve icon

Eerste voorbeeld is inderdaad wat ik bedoel, het tweede voorbeeld denk ik aan het volgende:

```html
<ol>
  <li>
    <span class="nl-unordered-list__inline-marker">
      <span class="nl-icon" aria-hidden="true">
        <svg />
      </span>
    </span>
    Foo Bar
  </li>
</ol>
```

Author: Netjes samengevat in deze issue comment https://github.com/nl-design-system/candidate/issues/1204#issuecomment-5251880488 we hebben 'm besproken in Candidate Go Go Go, en het is verwerkt in de HTML PR: https://github.com/nl-design-system/candidate/pull/1283

| `className`   | `string`                          | -       | nee       | Extra CSS classes naast `nl-ordered-list__item`.                                               |
| `ref`         | `Ref<HTMLLIElement>`              | -       | nee       | Ref naar het onderliggende `li` HTML-element.                                                  |
| `restProps`   | `LiHTMLAttributes<HTMLLIElement>` | -       | nee       | Overige standaard HTML-attributen en events worden doorgestuurd naar het `li` HTML-element.    |

#### Geldige combinaties van React OrderedListItem properties

| Configuratie                                                                     | Geldig | Opmerking                                                               |
| -------------------------------------------------------------------------------- | :----: | ----------------------------------------------------------------------- |
| `<OrderedListItem inlineMarker={<Icon />} inlineMarkerLabel="Stap 1">Item</OrderedListItem>` |   ✅   | Een inline marker moet een toegankelijke naam hebben.                   |

> | `<OrderedListItem inlineMarker={<Icon />}>Item</OrderedListItem>`                      |   ❌   | `inlineMarkerLabel` is verplicht wanneer een inline marker wordt gebruikt.    |

Reviewer: `inlineMarkerLabel` zou toch niet verplicht moeten zijn als er al een informatieve icon in zit met een alternatieve tekst?

Ik denk dat we twee situaties moeten supporten, en ik denk dat we daar enkele teksten voor moeten aanpassen die het hebben over informatieve icons.

1. Je kunt aan de ene kant een inline marker hebben met een informatieve icon met alt tekst. Dan ben je gelijk klaar. Bijvoorbeeld: `<OrderedListItem inlineMarker={<Icon label="1." />} />`
2. Aan de andere kant kun je een informatieve icon zonder alt tekst hebben waarbij we een convenience kunnen aanbieden om alsnog een label in te stellen via `inlineMarkerLabel`.
3. En dan nog de edge case van een informatieve icon met een verkeerde alt tekst (`<svg><title>Number One Icon</title>...</svg>` bijvoorbeeld) hebben waarbij we een convenience kunnen aanbieden om alsnog een beter label in te stellen via `inlineMarkerLabel`. Wanneer je `inlineMarkerLabel` instelt, dán wil je waarschijnlijk automatisch `aria-hidden="true"` op de icon container instellen.

Author: De drie scenarios zijn toegevoegd aan de Ordered List Testcases voor gebruik van de component: https://github.com/nl-design-system/candidate/issues/1191

In onderstaande comment is de historie over de juiste markup: https://github.com/nl-design-system/candidate/pull/1284#discussion_r3726755435
